app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt

```diff
@@ -310,72 +310,57 @@ class SavingAccountsTransactionFragment : BaseFragment() {
 
         tvStartDate?.text = DateHelper.getDateAsStringFromLong(startDate)
         tvEndDate?.text = DateHelper.getDateAsStringFromLong(endDate)
-        // setup listeners
+
         tvStartDate?.setOnClickListener { startDatePick() }
         tvEndDate?.setOnClickListener { endDatePick() }
-        checkBoxPeriod?.setOnClickListener {
-            checkBoxPeriod.isChecked = (!checkBoxPeriod.isChecked)
-        }
+
         checkBoxPeriod?.setOnCheckedChangeListener { _, isChecked ->
             isCheckBoxPeriod = isChecked
+            tvStartDate?.isEnabled = isChecked
+            tvEndDate?.isEnabled = false
+
             if (!isChecked) {
                 isReady = false
                 radioGroupFilter?.clearCheck()
                 selectedRadioButtonId = -1
             } else {
                 if (selectedRadioButtonId == -1) {
-                    val btn = dialogView.findViewById<RadioButton>(R.id.rb_date)
-                    btn.isChecked = true
+                    val btn = dialogView?.findViewById<RadioButton>(R.id.rb_date)
```

Comment:
why havent we used viewbinding here ? Use binding wherever needed

Reply:
what should i name the variable of binding..?